godot-macros/src/docs.rs

```diff
@@ -46,17 +46,17 @@ pub fn make_inherent_impl_docs(
     ) -> Option<TokenStream> {
         let methods = functions
             .iter()
-            .map(make_method_docs)
-            .collect::<Option<String>>()?;
+            .filter_map(make_method_docs)
+            .collect::<String>();
```

Comment:
theres a new problem here-- i implemented the option stuff so as to save space when no documentation is provided for a function, but now you just have empty documentation. you need to check if the number of documented items is 0 and return none to save space here

Reply:
Makes sense, fixed – method won't generate TokenStream in question if there is no need to do so

Reply:
its not when the `methods` are empty its when the methods have no documentation at all, i.e methods.iter().filter(has_docs).count() == 0

Reply:
you see when 
```rust
impl X {
   #[func] fn y(…) { }
}
```
we dont want to have a `<methods>` block

Reply:
you see when 
```rust
impl X {
   #[func] fn y(…) { }
}
```
we dont want to have a `<methods>` block

Reply:
Sure thing – right now (on master) if no methods are present, the generated xml contains methods tag `<methods></methods>` with no children present, and same goes for signals and constants. I'll just check with test project if Godot has no issues with missing "methods" block in case if there is _some_ documentation

Reply:
I dont remember why i put the `Option`s in at all then...

But if its working fine right now, you should be able to make the function just return Some, and thus remove the function from existence as it was a IIFE moved out (makeshift `try`)

Reply:
I took the more direct approach – I decided to keep all the declarations as an option and just handle it while generating docs.